core/escrow.js

```diff
@@ -3,16 +3,51 @@
 
 import harden from '@agoric/harden';
 import { mustBeSameStructure } from '../util/sameStructure';
+import { insist } from '../util/insist';
+
+const escrowExchange = harden({
+  // config is used for two customizations: 1) customization the names
+  // of the left and right seat, and 2) customizing the methods on the
+  // seats. For instance, in the case of an escrowed payment that has
+  // an associated use object, `makeCustomLeftSeatSrc` could be a
+  // stringified function that adds a `getUse` method to the seat to
+  // retrieve the use object for the escrowed payment.
+  start: async (
+    terms,
+    inviteMaker,
+    evaluate,
+    config = harden({
+      leftSeatName: 'left',
+      rightSeatName: 'right',
+      makeCustomLeftSeatSrc: `coreSeat => coreSeat`,
+      makeCustomRightSeatSrc: `coreSeat => coreSeat`,
+    }),
+  ) => {
+    const {
+      leftSeatName,
+      rightSeatName,
+      makeCustomLeftSeatSrc,
+      makeCustomRightSeatSrc,
+    } = config;
+
+    function evalStrToFn(source) {
+      insist(
+        typeof source === 'string',
+      )`"${source}" must be a string, but was ${typeof source}`;
+      const fn = evaluate(source, { harden, E });
+      insist(
+        typeof fn === 'function',
+      )`"${source}" must be a string for a function, but produced ${typeof fn}`;
+      return fn;
+    }
+
+    const makeCustomLeftSeat = evalStrToFn(makeCustomLeftSeatSrc);
+    const makeCustomRightSeat = evalStrToFn(makeCustomRightSeatSrc);
 
-// For clarity, the code below internally speaks of a scenario is which Alice is
-// trading some of her money for some of Bob's stock. However, for generality,
-// the API does not expose names like "alice", "bob", "money", or "stock".
-// Rather, Alice and Bob are left and right respectively. Money represents the
-// rights transferred from left to right, and Stock represents the rights
-// transferred from right to left.
-const escrowExchange = {
-  start: (terms, inviteMaker) => {
-    const { left: moneyNeeded, right: stockNeeded } = terms;
+    const {
+      [leftSeatName]: leftOfferAmount,
```

Comment:
I don't see how having custom seat names is making things simpler. Now the user of escrow has to define terms indexed by the new names, then specify the names when creating the escrow, and after spawning use the custom names to pull apart the invites.

And while customizing, they still have to deal with left and right seats and left and right escrows.

What's the benefit?

Reply:
Here's how I'm thinking about custom seat names. By default, 'left' and 'right' are used, so if the user doesn't want to define anything, they get 'left' and 'right' exactly as it has always been, so it's just as simple for the lazy user. For the user who wants to be able to provide custom names, such as 'alice' and 'bob' or 'offersStock' and 'offersMoney' it is much more understandable than 'left' and 'right'. I had to keep reminding myself of whether left was Alice, and if so, was Alice the one offering money. Now, I can define it be exactly what I want, and I will get back the seat names as I defined them. It's true that while they are in the moment of customization, a user will have to deal with setting the 'leftSeatName' and 'rightSeatName', but the benefit is that the user doesn't have to think about left and right ever again for the length of the contract.

Reply:
left and right are in again, and custom seat names are out